src/tunnel.cc

```diff
@@ -127,6 +127,12 @@ class TunnelStateData: public PeerSelectionInitiator
     /// called when negotiations with the peer have been successfully completed
     void notePeerReadyToShovel(const Comm::ConnectionPointer &);
 
+    /// \copydoc HierarchyLogEntry::stopPeering()
+    void stopPeering() {
+        Assure(request->hier.totalResponseTime().running());
```

Comment:
As I noted [elsewhere](https://github.com/measurement-factory/squid/pull/249/files/0a2b9b484adf99f220a4e3098ab6aa78be7eccd1#r1569051869), this check is unreliable, i.e., it does not protect TunnelStateData from calling stopPeering() twice. I think we s should remove it at all (and the wrapper method). Also we can add some additional TunnelStateData flag indicating whether stopPeering() has been called.